### PR TITLE
[ALLUXIO-1322][WIP] Prepared classes for MissingCtor rule in Checkstyle

### PR DESCRIPTION
**Pull request by HorizonNet**
https://alluxio.atlassian.net/browse/ALLUXIO-1322

Prepared some classes for the upcoming MissingCtor rule in Checkstyle. It checks that classes (except abstract ones) define a constructor and don't rely on the default one. The rule will look like the following

<module name="MissingCtor"/>

This ticket is WIP and needs some additional PRs.

### CODE REVIEW

File: core/common/src/main/java/alluxio/URI.java [hunk 1]
```diff
@@ -38,6 +38,10 @@
    * Factory for {@link URI}.
    */
   class Factory {
+
+    // prevent instantiation
+    private Factory() {}
```
Comment (jsimsa) [inline core/common/src/main/java/alluxio/URI.java]: `private Factory() {} // prevent instantiation`

File: core/common/src/main/java/alluxio/wire/ThriftUtils.java [hunk 2]
```diff
@@ -19,6 +19,9 @@
 @ThreadSafe
 public final class ThriftUtils {
 
+  // prevent instantiation
+  private ThriftUtils() {}
```
Comment (jsimsa) [inline core/common/src/main/java/alluxio/wire/ThriftUtils.java]: ditto

File: examples/src/main/java/alluxio/examples/keyvalue/ShowKeyValueStore.java [hunk 3]
```diff
@@ -41,6 +41,9 @@ private static void show(KeyValuePair pair, String scope) {
     }
   }
 
+  // prevent instantiation
+  private ShowKeyValueStore() {}
```
Comment (jsimsa) [inline examples/src/main/java/alluxio/examples/keyvalue/ShowKeyValueStore.java]: ditto

**Comment by jsimsa** [comment]
@HorizonNet thanks, left some minor comments

**Comment by HorizonNet** [comment]
@jsimsa Thanks for the comments. I updated the PR accordingly.

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**